### PR TITLE
Authentication and test changes for macos

### PR DESCRIPTION
**Pull request by yakman2020**
This enables macos to linux or windows connectivity with kerberos authentication. 

It does not implement http encryption from mac OS to windows..

### CODE REVIEW

**Comment by yakman2020** [comment]
@Microsoft/omi-devs  pls review.

File: Unix/http/httpclientauth.c [hunk 1]
```diff
@@ -9,9 +9,9 @@
 #include <config.h>
 #if AUTHORIZATION
 #if defined(macos)
-//#include <GSS/GSS.h>
-#include <Kerberos/Kerberos.h>
-#define HEIMDAL 1
+#include <GSS/GSS.h>
+//#include <Kerberos/Kerberos.h>
```
Comment (jeffaco) [inline Unix/http/httpclientauth.c]: Why is this commented out? Why not just get rid of it entirely?

File: Unix/http/httpclientauth.c [hunk 2]
```diff
@@ -41,8 +41,10 @@
 #define FORCE_TRACING  1
 
 #if GSS_USE_IOV
+#if !defined(macos)
```
Comment (jeffaco) [inline Unix/http/httpclientauth.c]: Can't lines 43 and 44 be combined into one, eliminating 47?

File: Unix/http/httpclientauth.c [hunk 3]
```diff
@@ -2028,17 +2035,15 @@ HttpClient_NextAuthRequest(_In_ struct _HttpClient_SR_SocketData * self, _In_ co
     const gss_OID_desc mechset_avail_elems[] = {
         { 6, "\053\006\001\005\005\002" },                  // Spnego
         { 10, "\x2b\x06\x01\x04\x01\x82\x37\x02\x02\x0a" }, // ntlm
-        { 9, "\052\206\110\206\367\022\001\002\002" },      // mech_krb5
-        { 6, "\053\006\001\005\002\005" }                   // mech_iakerb
+        { 9, "\052\206\110\206\367\022\001\002\002" }      // mech_krb5
```
Comment (jeffaco) [inline Unix/http/httpclientauth.c]: Given that this is now here, can 2029 be deleted? Come to think of it, can 2030 and 2032 be deleted? And 2033/2034?

Comment (yakman2020) [inline Unix/http/httpclientauth.c]: done

File: Unix/http/httpclientauth.c (hunk 2 above)
Comment (yakman2020) [inline Unix/http/httpclientauth.c]: done

File: Unix/http/httpclientauth.c (hunk 1 above)
Comment (yakman2020) [inline Unix/http/httpclientauth.c]: done

**Pull request merged**